File: frontend/src/api/ai.ts
import { http } from "./http";
import type { AiCoachResponse, AiProviderStatusResponse } from "../types/video";

export async function getAiCoachReport(pipelineId: string): Promise<AiCoachResponse> {
  const { data } = await http.post<AiCoachResponse>(`/api/ai/reports/${pipelineId}/coach`, {
    force_refresh: false,
  }, {
    timeout: 120_000,
  });
  return data;
}

export async function getAiProviderStatus(): Promise<AiProviderStatusResponse> {
  const { data } = await http.get<AiProviderStatusResponse>("/api/ai/status");
  return data;
}

File: backend/app/api/ai.py
from __future__ import annotations

from fastapi import APIRouter

from app.schemas.dto import AiCoachRequest, AiCoachResponse, AiProviderStatusResponse
from app.services.ai_coach import ai_provider_status, generate_ai_coach
from app.services.pipeline import get_pipeline_result_summary

router = APIRouter(prefix="/ai", tags=["ai"])


@router.post("/reports/{pipeline_id}/coach", response_model=AiCoachResponse)
async def ai_report_coach_api(pipeline_id: str, body: AiCoachRequest | None = None):
    _ = body
    summary = get_pipeline_result_summary(pipeline_id)
    return AiCoachResponse(**generate_ai_coach(summary))


@router.get("/status", response_model=AiProviderStatusResponse)
async def ai_status_api():
    return AiProviderStatusResponse(**ai_provider_status())
